File: src/Services/Products/Products.Core/Product.cs
namespace Products.Core;

public class Product
{
    public Guid Id { get; set; }
    public string Barcode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Brand { get; set; }
    public string? Category { get; set; }
    public DateTime ScannedAt { get; set; } = DateTime.UtcNow;
}

File: NutritionLabelScanner/src/NutritionLabelScanner.Web/projects/api/src/lib/models/product.ts
export interface Product {
  productId: string;
  userId: string;
  name: string;
  brand?: string;
  barcode?: string;
  category?: string;
  servingSize?: string;
  scannedAt: string;
  notes?: string;
  createdAt: string;
}

export interface CreateProduct {
  userId: string;
  name: string;
  brand?: string;
  barcode?: string;
  category?: string;
  servingSize?: string;
  scannedAt: string;
  notes?: string;
}

export interface UpdateProduct {
  productId: string;
  userId: string;
  name: string;
  brand?: string;
  barcode?: string;
  category?: string;
  servingSize?: string;
  scannedAt: string;
  notes?: string;
}
